### PR TITLE
This closes #889 , XmlSuite in nested directories results in FIleNotFoundException

### LINKED ISSUE #889: XmlSuite in nested directories results in FIleNotFoundException
Basically I want to add XmlSuite directly to TestNG and then run. Also, my XmlSuite is a suite-files which contains two more suite files.

``` xml
    <suite name="suite of suites">
    <suite-files>
        <suite-file path="TestNG2.xml" />
        <suite-file path="TestNG.xml" />
    </suite-files>
</suite>
```

Till the time main suite file and the child suite files are in the root folder everything works fine. However, if these files are in nested folder then we will get `FileNotFoundException` in `TestNG.initializeSuitesAndJarFile` at line 279 -> `Collection<XmlSuite> childSuites = getParser(suiteFile).parse();`

Here is the relevant part of the code

``` java
for (XmlSuite s: m_suites) {

        for (String suiteFile : s.getSuiteFiles()) {
            try {
                Collection<XmlSuite> childSuites = getParser(suiteFile).parse();
                for (XmlSuite cSuite : childSuites){
                    cSuite.setParentSuite(s);
                    s.getChildSuites().add(cSuite);
                }
```

Here we are passing simple the file name which will be the file name mentioned as the path attribute of    `<suite-file path="TestNG2.xml" />` node in xml file.

If the suite is located inside a child directory of project Root folder, it will result in `FileNotFoundException`

Fix: We have to pass the full path to parser.

### PR DESCRIPTION
**Pull request by virenv**

### CODE REVIEW

File: src/main/java/org/testng/TestNG.java [hunk 1]
```diff
@@ -276,10 +278,11 @@ public void initializeSuitesAndJarFile() {
 
         for (String suiteFile : s.getSuiteFiles()) {
             try {
-                Collection<XmlSuite> childSuites = getParser(suiteFile).parse();
+            	Path rootPath = Paths.get(s.getFileName()).getParent();
+		        Collection<XmlSuite> childSuites = getParser(rootPath.resolve(suiteFile).toString()).parse();
                 for (XmlSuite cSuite : childSuites){
-                    cSuite.setParentSuite(s);
-                    s.getChildSuites().add(cSuite);
+	                cSuite.setParentSuite(s);
```
Comment (cbeust) [inline src/main/java/org/testng/TestNG.java]: Unindent.

File: src/main/java/org/testng/TestNG.java [hunk 2]
```diff
@@ -276,10 +278,11 @@ public void initializeSuitesAndJarFile() {
 
         for (String suiteFile : s.getSuiteFiles()) {
             try {
-                Collection<XmlSuite> childSuites = getParser(suiteFile).parse();
+            	Path rootPath = Paths.get(s.getFileName()).getParent();
+		        Collection<XmlSuite> childSuites = getParser(rootPath.resolve(suiteFile).toString()).parse();
```
Comment (cbeust) [inline src/main/java/org/testng/TestNG.java]: Unindent.

**Comment by cbeust** [comment]
LGTM after the minor reformats.

**Comment by virenv** [comment]
Fixed the formatting issues. Some problem with command line, its always pushing in extra indented code.

File: src/main/java/org/testng/TestNG.java [hunk 3]
```diff
@@ -276,10 +278,11 @@ public void initializeSuitesAndJarFile() {
 
         for (String suiteFile : s.getSuiteFiles()) {
             try {
-                Collection<XmlSuite> childSuites = getParser(suiteFile).parse();
+            	 Path rootPath = Paths.get(s.getFileName()).getParent();
```
Comment (juherr) [inline src/main/java/org/testng/TestNG.java]: If you declare this line outside the loop, you will compute it only one time.

Comment (virenv) [inline src/main/java/org/testng/TestNG.java]: Something seriously wrong with my fork. Let me delete and re create a fork and come back with this change.

**Pull request closed without merging**